crispr_sites/crispr_sites.cpp

```diff
@@ -338,30 +343,51 @@ void scan_stdin() {
     // and then merging incrementally with c++ algorithm set_union,
     // rather than doing a huge sort at the end.   Parallelizing, esp on GPU,
     // could yield phenomenal speedup if we ever need to run this program fast.
+
+    // Counting the multiplicity of guides is used for DASH guide
+    // creation in dashdat -dynerman
+    
     cerr << "Sorting " << results.size() << " candidate guides." << endl;
     sort(results.begin(), results.end());
 
     // 0 is not a valid code
     int64_t last = 0;
+    int32_t count = 0;
     for (auto it = results.begin();  it != results.end();  ++it) {
         if (*it != last) {
             ++guides;
+
+            results_unique.push_back(make_tuple(*it, count));
+            count = 0;
         }
         last = *it;
+	++count;
     }
 
     cerr << "Outputting " << guides << " unique guides." << endl;
     last = 0;
     char obuf[k-1];
     obuf[k-2] = 0;
     obuf[k-3] = '\n';
-    for (auto it = results.begin();  it != results.end();  ++it) {
-        if (*it != last) {
-            decode(obuf, k-3, *it);
-            cout << obuf;
+
+    if (output_counts) {
+        obuf[k-3] = '\t';
+    }
+    
+    for (auto it = results_unique.begin();  it != results_unique.end();  ++it) {
+        decode(obuf, k-3, get<0>(*it));
+        cout << obuf;
+        if (output_counts) {
+            cout << get<1>(*it) << endl;
         }
-        last = *it;
     }
+    // for (auto it = results.begin();  it != results.end();  ++it) {
+    //     if (*it != last) {
+    //         decode(obuf, k-3, *it);
+    //         cout << obuf;
+    //     }
+    //     last = *it;
+    // }
```

Comment:
You can delete this commented out code.   There is github to show older versions. :)